File: src/main/java/rbq/wtf/lycoris/client/detector/MargeleAntiCheatDetector.java
package rbq.wtf.lycoris.client.detector;

public class MargeleAntiCheatDetector {
    public static boolean isMAC() {
        return getMAC() != null;
    }

    public static Class<?> getMAC() {
        try {
            return Class.forName("cn.margele.netease.clientside.MargeleAntiCheat");
        } catch (ClassNotFoundException e) {
            return null;
        }
    }

    public static Class<?> getHyGui() {
        try {
            return Class.forName("cn.hycraft.core.gui.HyCraftGui");
        } catch (ClassNotFoundException e) {
            return null;
        }
    }

    public static Class<?> getGuiTab() {
        try {
            return Class.forName("cn.hycraft.core.util.render.EnumUtils$GuiTab");
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
            return null;
        }
    }

    public static Enum<?> getHyTab() {
        try {
            return (Enum<?>) Class.forName("cn.hycraft.core.util.render.EnumUtils$GuiTab").getEnumConstants()[0];
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: src/main/java/rbq/wtf/lycoris/client/detector/MargeleAntiCheatDetector.kt
package rbq.wtf.lycoris.client.detector

object MargeleAntiCheatDetector {
    @JvmStatic
    val isMAC: Boolean
        get() = getMAC() != null

    @JvmStatic
    fun getMAC(): Class<*>? {
        return try {
            this.javaClass.classLoader.loadClass("cn.margele.netease.clientside.MargeleAntiCheat")
        } catch (e: ClassNotFoundException) {
            null
        }
    }

    @JvmStatic
    val hyGui: Class<*>?
        get() = try {
            this.javaClass.classLoader.loadClass("cn.hycraft.core.gui.HyCraftGui")
        } catch (e: ClassNotFoundException) {
            null
        }

    @JvmStatic
    val guiTab: Class<*>?
        get() = try {
            this.javaClass.classLoader.loadClass("cn.hycraft.core.util.render.EnumUtils\$GuiTab")
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
            null
        }

    @JvmStatic
    val hyTab: Enum<*>?
        get() = try {
            this.javaClass.classLoader.loadClass("cn.hycraft.core.util.render.EnumUtils\$GuiTab").enumConstants[0] as Enum<*>
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
}